### DISCUSSION TITLE
Resizable Sidebar

### DISCUSSION THREAD

**Discussion by sophiakanoa** [facebook/docusaurus#9676] [Q&A]
I am hoping to make my sidebar resizable by having a divider that the user can drag to make the sidebar larger or smaller. Are there any ideas on ways to swizzle and create this? 
I am hoping for a sidebar like the one in this sandbox: https://codesandbox.io/p/sandbox/react-resizable-sidebar-kz9de?file=%2Fsrc%2Findex.js%3A8%2C15

**Comment by Advait251206** [comment]
I think the cleanest way to achieve a resizable sidebar in Docusaurus is to wrap the sidebar and main layout in a split-pane style container and then control the sidebar width via React state.

A simple approach is:

- Store sidebar width in state (e.g. `useState(250)`).
- Add a vertical draggable divider between sidebar and content.
- Update width dynamically on `mousemove`.
- Persist value in `localStorage` if you want it to survive reloads.

If you're using swizzling, you can override `@docusaurus/theme-classic` layout and inject a resizable wrapper around the main layout component.

A minimal idea looks like:

- Attach `onMouseDown` to the divider
- Track dragging with `window.addEventListener('mousemove')`
- Clamp width between min/max values
- Remove listener on `mouseup`

Libraries like `react-resizable` or `re-resizable` can also simplify this a lot and save you from manually handling drag logic.

If you want behavior similar to the CodeSandbox you linked, using a library is usually the most stable option because it handles edge cases like selection dragging, touch support, and window resize automatically.

---

If this answer helped or pointed you in the right direction, I'd appreciate it if you could mark it as the accepted answer so it's easier for others with the same issue to find.

Also, if you found my contribution useful, I'd appreciate it if you could check out my GitHub profile, follow me, and star any repositories you find interesting.

GitHub: https://github.com/Advait251206
```
